examples/qml_minimal/CMakeLists.txt

```diff
@@ -78,20 +82,37 @@ add_library(${APP_NAME}_lib INTERFACE)
 # overwriting another's files.
 target_include_directories(${APP_NAME}_lib INTERFACE "${CXXQT_EXPORT_DIR}/${CRATE}")
 
-target_link_libraries(${APP_NAME}_lib INTERFACE
-    # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
-    # otherwise the linker will discard the static variables that initialize it.
-    "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"
-    Qt::Core
-    Qt::Gui
-    Qt::Qml
-    Qt::QuickControls2
-)
+if(BUILD_WASM)
+    target_link_libraries(${APP_NAME}_lib INTERFACE
+        Qt::Core
+        Qt::Gui
+        Qt::Qml
+        Qt::QuickControls2
+    )
+    
+    target_link_options(${APP_NAME}_lib INTERFACE 
+        "SHELL:-Wl,--whole-archive $<TARGET_FILE:${CRATE}-static> -Wl,--no-whole-archive"
+    )
+else()
+    target_link_libraries(${APP_NAME}_lib INTERFACE
+        # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
+        # otherwise the linker will discard the static variables that initialize it.
+        "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"
+        Qt::Core
+        Qt::Gui
+        Qt::Qml
+        Qt::QuickControls2
+    )
+endif()
```

Comment:
So found out how to tell CMake how to link when it is in an unfamiliar build env (originally from https://discourse.cmake.org/t/error-when-crosscompiling-with-whole-archive-target-link/9394/3 ).

This results in a much simpler code as we can keep the original `target_link_libraries` and have a short difference for `BUILD_WASM`.

Note that you could probably put this if block higher up in the example cmake files so that it isn't shown in the general book pages and then have a separate book anchor for the wasm pages to show how you need to specify this when using wasm ?

Or maybe it even just goes globally in the top CMake file then you document in the book that you need to add those two lines for wasm builds by using a book anchor ?

Note this also solves the double build issue as the dependency is correctly done by `target_link_libraries` now

```cmake
# When building under emscripten tell CMake how to link using whole archive
if(BUILD_WASM)
    set(CMAKE_LINK_LIBRARY_USING_WHOLE_ARCHIVE "-Wl,--whole-archive" "<LINK_ITEM>" "-Wl,--no-whole-archive")
    set(CMAKE_LINK_LIBRARY_USING_WHOLE_ARCHIVE_SUPPORTED True)
endif()

target_link_libraries(${APP_NAME}_lib INTERFACE
    # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
    # otherwise the linker will discard the static variables that initialize it.
    "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"
    Qt::Core
    Qt::Gui
    Qt::Qml
    Qt::QuickControls2
)
```